Aplib.Core/IDocumented.cs

```diff
@@ -0,0 +1,16 @@
+namespace Aplib.Core
+{
+    /// <summary>
+    /// Represents an object that contains an instance of <see cref="IMetadata"/>.
+    /// </summary>
+    public interface IDocumented
+    {
+        /// <summary>
+        /// Gets the metadata of this BDI cycle component.
```

Comment:
Would `IDocumented` only be scoped at BDI cycle components. What if we add multiple agents, and we want to add documentation to those agents?

Reply:
The same goes for `IMetadata`: Try to provide descriptions to the interfaces that are as broad as possible. This way, the documentation could be inherited for any class that chooses to implement this interface.